stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheetPaymentMethodsListFragment.kt

```diff
@@ -12,52 +12,62 @@ import com.stripe.android.databinding.FragmentPaymentsheetPaymentMethodsListBind
         PaymentSheetViewModel.Factory {
             requireActivity().application
         }
     }
 
-    private val fragmentViewModel by viewModels<VM>()
+    private val fragmentViewModel by viewModels<PaymentMethodsViewModel>()
+
+    private val adapter: PaymentSheetPaymentMethodsAdapter by lazy {
+        PaymentSheetPaymentMethodsAdapter(
+            fragmentViewModel.selectedPaymentMethod,
+            paymentMethodSelectedListener = {
+                fragmentViewModel.selectedPaymentMethod = it
+                activityViewModel.updateSelection(it)
+            },
+            addCardClickListener = {
+                activityViewModel.transitionTo(
+                    PaymentSheetViewModel.TransitionTarget.AddPaymentMethodFull
+                )
+            }
+        )
+    }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        if (activity == null) {
-            return
-        }
+        val activity = this.activity ?: return
 
-        // If we're returning to this fragment from elsewhere, we need to reset the selection to whatever
-        // the user had selected previously
+        // If we're returning to this fragment from elsewhere, we need to reset the selection to
+        // whatever the user had selected previously
         activityViewModel.updateSelection(fragmentViewModel.selectedPaymentMethod)
         // reset the mode in case we're returning from the back stack
         activityViewModel.updateMode(SheetMode.Wrapped)
 
         val binding = FragmentPaymentsheetPaymentMethodsListBinding.bind(view)
-        binding.recycler.layoutManager = LinearLayoutManager(requireContext(), LinearLayoutManager.HORIZONTAL, false)
+        binding.recycler.layoutManager = LinearLayoutManager(
+            activity,
+            LinearLayoutManager.HORIZONTAL,
+            false
+        )
+        binding.recycler.adapter = adapter
 
         activityViewModel.paymentMethods.observe(viewLifecycleOwner) { paymentMethods ->
-            binding.recycler.adapter = PaymentSheetPaymentMethodsAdapter(
-                paymentMethods,
-                fragmentViewModel.selectedPaymentMethod,
-                paymentMethodSelectedListener = {
-                    fragmentViewModel.selectedPaymentMethod = it
-                    activityViewModel.updateSelection(it)
-                },
-                addCardClickListener = {
-                    activityViewModel.transitionTo(PaymentSheetViewModel.TransitionTarget.AddPaymentMethodFull)
-                }
-            )
+            adapter.paymentMethods = paymentMethods
         }
 
         // Only fetch the payment methods list if we haven't already
         if (activityViewModel.paymentMethods.value == null) {
-            activityViewModel.updatePaymentMethods(requireActivity().intent)
+            activityViewModel.updatePaymentMethods(activity.intent)
```

Comment:
Not something for this PR but this should actually probably be an error. I added this code before PaymentSheetLoadingFragment existed. Loading payment methods should happen there now so we should never instantiate `PaymentSheetPaymentMethodsListFragment` if `paymentMethods` is null or empty